### ISSUE TITLE
Import order with monkey.patch_all()

### ISSUE THREAD

**Issue by alanwilter** [astral-sh/ruff#21040]
### Question

Hi, I know `ruff` is aware of:

```python
import matplotlib

matplotlib.use("Agg")

import matplotlib.pyplot as plt
```

So it does not raise a false alarm here.

But what about `monkey.patch_all()`?

AFAIK it's said the this must be executed as early as possible, so, e.g.:

```python
from gevent import monkey

monkey.patch_all()

import psycogreen.gevent
```

But then `ruff` throws `E402`.

Is it a bug? Am I outdated or am I missing something else?

### Version

ruff 0.14.0

**Comment by ntBre** [comment]
There have been a few related issues around E402 but not for this library specifically. #19904 probably has the most discussion. In short, this is a known limitation, but we're still considering whether we want to make a change here or just recommend a `noqa` comment when this pattern is needed.
